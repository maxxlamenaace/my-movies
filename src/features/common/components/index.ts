import Loader from './loader/Loader';
import Logo from './logo/Logo';
import Footer from './footer/Footer';
import AppBar from './app-bar/AppBar';
import Container from './container/Container';
import ScrollableAppBar from './scrollable-app-bar/ScrollableAppBar';
import Sidebar from './sidebar/Sidebar';
import ManualSwiper from './manual-swiper/ManualSwiper';
import Wrapper from './wrapper/Wrapper';
import PaginatorSwiper from './paginator-swiper/PaginatorSwiper';

export {
  Loader,
  Logo,
  Footer,
  AppBar,
  Container,
  ScrollableAppBar,
  Sidebar,
  ManualSwiper,
  Wrapper,
  PaginatorSwiper,
};
