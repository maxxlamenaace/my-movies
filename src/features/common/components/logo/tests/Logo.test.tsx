import React from 'react';

import { BrowserRouter } from 'react-router-dom';
import { render, cleanup, act, fireEvent, waitFor } from '@testing-library/react';

import { Logo } from '@/features/common';

describe('Logo', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders correctly', () => {
    const { asFragment } = render(<Logo />, { wrapper: BrowserRouter });
    expect(asFragment()).toMatchSnapshot();
  });

  describe('when user clicks on logo', () => {
    it('redirects to the root page', () => {
      const { container } = render(<Logo />, {
        wrapper: BrowserRouter,
      });

      act(() => {
        const item = container.querySelector('#logo-link');
        if (item) {
          fireEvent.click(item);
        }
      });

      waitFor(() => {
        expect(global.window.location.href).toEqual('/');
      });
    });
  });
});
