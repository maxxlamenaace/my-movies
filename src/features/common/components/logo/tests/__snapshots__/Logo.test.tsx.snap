// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly 1`] = `
<DocumentFragment>
  <a
    href="/"
    id="logo-link"
    style="text-decoration: none;"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-10bvdjj-MuiTypography-root"
      id="logo"
      style="color: rgba(0, 0, 0, 0.87);"
    >
      my
      <span
        style="color: rgb(25, 118, 210);"
      >
        Movies
      </span>
    </p>
  </a>
</DocumentFragment>
`;
